### PR TITLE
added a bunch of recipes

### PR DESCRIPTION
**Pull request by ericmjl**

### CODE REVIEW

**Comment by patricksnape** [comment]
Obviously the linter is unhappy about a load of these recipes - furthermore - we generally prefer each PR to contain a single recipe unless the recipes are co-dependant. Here, however, I think this appears to be a bag of packages you would like included on conda-forge?

File: recipes/autograd/bld.bat [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"%PYTHON%" setup.py install
+if errorlevel 1 exit 1
+
+:: Add more build steps here, if they are necessary.
```
Comment (patricksnape) [inline recipes/autograd/bld.bat]: You can remove all these commented lines.

File: recipes/autograd/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    # syntax is module:function.  For example
+    #
+    # - autograd = autograd:main
+    #
+    # Would create an entry point called autograd that calls autograd.main()
+
+
+  # If this is a new build for the same version, increment the build
+  # number. If you do not include this key, it defaults to 0.
+  number: 0
```
Comment (patricksnape) [inline recipes/autograd/meta.yaml]: You can replace the two build scripts with `script: python setup.py install --single-version-externally-managed --record=record.txt`

File: recipes/autograd/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  summary: 'Efficiently computes derivatives of numpy code.'
+
+# See
+# http://docs.continuum.io/conda/build.html for
+# more information about meta.yaml
+
+extra:
+  recipe-maintainers:
+    # Always check with the people listed below if they are OK becoming maintainers of the recipe. (There will be spam!)
+    - ericmjl
```
Comment (patricksnape) [inline recipes/autograd/meta.yaml]: Missing terminal new line

File: recipes/autograd/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
+package:
+  name: autograd
+  version: "1.1.5"
```
Comment (patricksnape) [inline recipes/autograd/meta.yaml]: We usually put the version in a jinja tag at the top, see above.

File: recipes/autograd/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+source:
+  fn: autograd-1.1.5.tar.gz
+  url: https://files.pythonhosted.org/packages/e5/70/e9c8db296d6ff33a17fec0ecaf11e3efbc85722d042365d01704046befef/autograd-1.1.5.tar.gz
```
Comment (patricksnape) [inline recipes/autograd/meta.yaml]: We prefer the `pypi.io` url, something like the following (assuming the use of the jinja tags):

```
source:
  fn: {{ name }}-{{ version }}.tar.gz
  url: https://pypi.python.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
  md5: MD5_HERE
```

File: recipes/autograd/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
+package:
```
Comment (patricksnape) [inline recipes/autograd/meta.yaml]: Above this we generally have a couple of jinja tags:

```
{% set name = "autograd" %}
{% set version = "1.1.5" %}
```

File: recipes/autograd/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+source:
+  fn: autograd-1.1.5.tar.gz
+  url: https://files.pythonhosted.org/packages/e5/70/e9c8db296d6ff33a17fec0ecaf11e3efbc85722d042365d01704046befef/autograd-1.1.5.tar.gz
+  md5: 2b65978109361a6ffb6986d03aebc157
+#  patches:
+   # List any patch files here
+   # - fix.patch
+
+build:
+  # noarch_python: True
```
Comment (patricksnape) [inline recipes/autograd/meta.yaml]: Please remove all the comments from the template.

File: recipes/circos/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  fn: Circos-1.3.5.tar.gz
+  url: https://files.pythonhosted.org/packages/fe/30/966e9d8ca0c8bb63bcefc1147ecf16f5518087c54b7a71cdc43920d9ae67/Circos-1.3.5.tar.gz
+  md5: a771f2ac0298bce6e4b1d1c20de35109
+#  patches:
+   # List any patch files here
+   # - fix.patch
+
+build:
+  # noarch_python: True
+  preserve_egg_dir: True
```
Comment (patricksnape) [inline recipes/circos/meta.yaml]: Is this required?

File: recipes/circos/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,72 @@
+package:
```
Comment (patricksnape) [inline recipes/circos/meta.yaml]: As in the autograd recipe, we would rather use the `pypi.io` url scheme + jinja tags.

File: recipes/circos/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    - numpy
+    - matplotlib
+    - setuptools
+    - pycodestyle
+
+  run:
+    - python
+    - numpy
+    - matplotlib
+    - setuptools
```
Comment (patricksnape) [inline recipes/circos/meta.yaml]: Is setuptools required at runtime?

File: recipes/conda_build.py [hunk 11]
```diff
@@ -0,0 +1,39 @@
+import os
```
Comment (patricksnape) [inline recipes/conda_build.py]: I'm confused what this script is for?

**Comment by patricksnape** [comment]
Please remove the template comments.
Please update recipes to use the `script` tag rather than separate `bld.bat` and `build.sh`.
Please use the `pypi.io` endpoint for downloading the packages (preferably using the jinja tag form).

File: recipes/conda_build.py (hunk 11 above)
Comment (ericmjl) [inline recipes/conda_build.py]: This is my own script, I've been keeping it in the directory for automatically downloading packages from PyPI and uploading them to my channel on `anaconda.org`.

**Comment by ericmjl** [comment]
@patricksnape: out of curiosity, why is it that `conda-forge` prefers the `script` tag rather than the automatically generated `bld.bat` and `build.sh` scripts that are generated from `conda skeleton pypi [pkgname]`?

Also, is there a tool that automatically formats the `meta.yml` file correctly for `conda-forge` based on package data? I have been doing `conda skeleton pypi [pkgname]` followed by `conda build [pkgname]`, and the `meta.yml` file is generated automatically. It is much less work if there's a way that this could be done.

**Comment by ericmjl** [comment]
@patricksnape: btw, in line with what is common practice for the community, I will close this PR and re-submit individual recipes.

**Comment by ocefpaf** [comment]
> @patricksnape: out of curiosity, why is it that `conda-forge` prefers the `script` tag rather than the automatically generated `bld.bat` and `build.sh` scripts that are generated from `conda skeleton pypi [pkgname]`?

That is for simplicity. A single file versus three files. Also, if you need to change the build script later you change it in only one place. BTW that may change in `conda-skeleton` soon.

To answer your second questions: we don't have any special tools to format the recipe, but we are planning on improving conda-skeleton.

> @patricksnape: btw, in line with what is common practice for the community, I will close this PR and re-submit individual recipes.

Thanks!

**Comment by patricksnape** [comment]
Thanks @ericmjl!

**Pull request closed without merging**